src/provider.tsx

```diff
@@ -0,0 +1,268 @@
+import {
+  ActionBar,
+  Button,
+  IconButton,
+  Portal,
+  useFileUpload,
+} from "@chakra-ui/react";
+import { AsyncDuckDBConnection } from "@duckdb/duckdb-wasm";
+import { useDuckDb } from "duckdb-wasm-kit";
+import {
+  useEffect,
+  useReducer,
+  useState,
+  type Dispatch,
+  type ReactNode,
+} from "react";
+import { LuFocus, LuX } from "react-icons/lu";
+import type { StacCollection } from "stac-ts";
+import { useStacCollections, useStacValue } from "./components/stac/hooks";
+import {
+  useStacGeoparquetItem,
+  useStacGeoparquetMetadata,
+  useStacGeoparquetTable,
+} from "./components/stac/stac-geoparquet";
+import { getCollectionsExtent } from "./components/stac/utils";
+import { toaster } from "./components/ui/toaster";
+import { StacMapContext, type SelectedCollectionsAction } from "./context";
+import { useFitBbox } from "./hooks";
+
+export function StacMapProvider({ children }: { children: ReactNode }) {
+  const [href, setHref] = useState<string | undefined>(getInitialHref());
+  const fileUpload = useFileUpload({ maxFiles: 1 });
+  const {
+    value,
+    loading: valueLoading,
+    parquetPath,
+    error: valueError,
+  } = useStacValue(href, fileUpload);
+  const {
+    collections,
+    loading: collectionsLoading,
+    error: collectionsError,
+  } = useStacCollections(value);
+  const [selectedCollections, selectedCollectionsDispatch] = useReducer(
+    selectedCollectionsReducer,
+    new Set<string>(),
+  );
+  const { connection, error: duckDbConnectionError } = useDuckDbConnection();
```

Comment:
I like how this makes it much easier to share a DuckDB connection throughout the whole app.